src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -516,61 +536,72 @@ def reconstruct_messages(
                     new_messages.append(Message(message.value.replace(kafka_payload)))
 
         with metrics.timer("metrics_consumer.reconstruct_messages.emit_payload_metrics"):
-            for use_case_id in self.__message_count:
-                metrics.incr(
-                    "metrics_consumer.process_message.messages_seen",
-                    amount=self.__message_count[use_case_id],
-                    tags={"use_case_id": use_case_id.value},
-                )
-                metrics.distribution(
-                    "metrics_consumer.process_message.message.avg_size_in_batch",
-                    self.__message_bytes[use_case_id] / self.__message_count[use_case_id],
-                    tags={"use_case_id": use_case_id.value},
-                    unit="byte",
-                )
-                metrics.distribution(
-                    "metrics_consumer.process_message.message.avg_tags_len_in_batch",
-                    self.__message_tags_len[use_case_id] / self.__message_count[use_case_id],
-                    tags={"use_case_id": use_case_id.value},
-                    unit="int",
-                )
-                metrics.distribution(
-                    "metrics_consumer.process_message.message.avg_value_len_in_batch",
-                    self.__message_value_len[use_case_id] / self.__message_count[use_case_id],
-                    tags={"use_case_id": use_case_id.value},
-                    unit="int",
-                )
+
+            # calculate the total number of parseable messages
+            num_parseable_messages = sum(self.__parseable_message_count.values())
+
+            # we have this check to detect the case that ALL of the messages
+            # in a batch are just invalid JSON and cannot be parsed.
+            # in this scenario, recording these metrics is not meaningful.
+            if num_parseable_messages > 0:
```

Comment:
> in this scenario, recording these metrics is not meaningful.

those are mostly performance related metrics. why are invalid messages not relevant to indexer performance? I would move the increment of the counter at Line 541 to the beginning of this function so this conditional is not necessary

Reply:
If you look at the way these metrics are calculated, you can see they rely on instance variables such as `self.__message_value_len`. Stuff like `self.__message_value_len` requires the payload to be at least JSON-parseable (i.e. it can't be completely invalid JSON).

Reply:
oh right, all of them depend on usecase. all good then i guess. still a bummer IMO

Reply:
Just to be clear, these metrics are still being recorded on other kinds of invalid messages (those that fail schema validation, and those that fail other kinds of checks)

Ideally, I think we could have some of these checks for "is this reasonable JSON" moved to Relay. I feel that if it is incomprehensible data, it doesn't quite make sense to send it to the indexer / have it collect metrics on it.

Reply:
> Ideally, I think we could have some of these checks for "is this reasonable JSON" moved to Relay. I feel that if it is incomprehensible data, it doesn't quite make sense to send it to the indexer / have it collect metrics on it.

right, I guess part of the problem though is that services disagree about what valid messages are.

Reply:
Yeah, this is something I'm interested in investigating/bringing up more.